web/app/scripts/resources/polygonstate/polygonstate-service.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Polygon Type state control - changes to the private vars which define this state
+ *  are broadcast to rootscope for use by controllers
+ */
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function PolygonState($window, $log, $rootScope, Polygons) {
+        var defaultParams, selected, options;
+        var _ = $window._;
+        this.updateOptions = updateOptions;
```

Comment:
For consistency with our typical angular style, consider using the format:
```
var module = {
updateOptions: updateOptions,
// ...
};
return module;
```
Instead of this + that.